firefox_ui_tests/functional/locationbar/test_star_in_autocomplete.py

```diff
@@ -19,44 +19,48 @@ class TestStarInAutocomplete(FirefoxTestCase):
     def setUp(self):
         FirefoxTestCase.setUp(self)
 
-        self.clear_history()
+        self.test_urls = [self.marionette.absolute_url('layout/mozilla_grants.html')]
 
         # Location bar suggests 'History and Bookmarks'
         self.prefs.set_pref(self.PREF_LOCATION_BAR_SUGGEST, 0)
 
-    def tearDown(self):
-        FirefoxTestCase.tearDown(self)
-
-        self.prefs.reset_pref(self.PREF_LOCATION_BAR_SUGGEST)
-
-    def clear_history(self):
         with self.marionette.using_context('content'):
             self.marionette.navigate('about:blank')
 
-        self.marionette.execute_script('''
-            let count = gBrowser.sessionHistory.count;
-            gBrowser.sessionHistory.PurgeHistory(count);
-        ''')
+        self.places.remove_all_history()
+
+    def tearDown(self):
+        # Close the autocomplete results
+        try:
+            self.browser.navbar.locationbar.autocomplete_results.close()
+            self.places.restore_default_bookmarks()
+        finally:
+            FirefoxTestCase.tearDown(self)
 
     @skip_under_xvfb
     def test_star_in_autocomplete(self):
         search_string = 'grants'
 
+        def visit_urls():
+            for url in self.test_urls:
+                self.marionette.navigate(url)
+
         with self.marionette.using_context('content'):
```

Comment:
The with statement you better have to put into the callback. The helper `wait_for_visited` is working in chrome scope.